test/test_config.py

```diff
@@ -102,3 +102,6 @@ def test_config_container_registry(self):
         test_config["container_registry"]["owner"] = ""
         config.init_global(test_config)
         assert (config.config.container_prefix == config.DEFAULT_CONTAINER_PREFIX)
+
+    def test_dataset_label(self):
```

Comment:
We don't need to test everything. Test coverage should make sure that ~2 valid labels pass and 2-3 invalid labels raise ValueErrors. Example invalid labels can contain `-`, another special character like `$`, and a Unicode symbol like `✉`.

We'll need to think about the most efficient way to test these. Do we need to write separate .yml files for each, or can we use a `sed` like operation to create temp copies of the `config.yml` file with modified dataset labels and then parse those?

Reply:
I tried something similar to the functions above in the file. It seems like I can just overwrite the temp_config and initialize the config to trigger the error.